httpx/dispatch/connection_pool.py

```diff
@@ -106,13 +109,26 @@ def max_connections(self) -> BasePoolSemaphore:
     def num_connections(self) -> int:
         return len(self.keepalive_connections) + len(self.active_connections)
 
+    async def keepalive_timeouts(self) -> None:
+        now = self.backend.time()
+        if now < self.next_keepalive_check:
+            return
+        self.next_keepalive_check = now + 1.0
+        keepalives = list(self.keepalive_connections.all.keys())
```

Comment:
Might be worth a comment to indicate that we're loading these into a list to prevent "changed during iteration" issues (which I assume is why we're doing this in the first place?)?

Reply:
Indeed.

Reply:
Also perhaps worth using a tuple for [true immutability](https://docs.python.org/3/tutorial/datastructures.html#tuples-and-sequences)?

Reply:
@stephenbrown2 I don’t think that’s necessary, because even if we mutate the list (which we never actually do here) then it’s already separate from the connection set so it shouldn’t have any incidence.

Reply:
Agreed, it's a "private" variable only used for iteration, so not truly necessary to make it a tuple.